trickler/scales.py

```diff
@@ -40,6 +40,133 @@ def noop(*args, **kwargs):
+        OVERLOAD = 2
+        ERROR = 3
+        MODEL_NUMBER = 4
+        SERIAL_NUMBER = 5
+        ACKNOWLEDGE = 6
+
+    class ScaleStatusV2(enum.Enum):
+        """New version avoids zero (0) which can be confused with null/None."""
+        STABLE = 1
+        UNSTABLE = 2
+        OVERLOAD = 3
+        ERROR = 4
+        MODEL_NUMBER = 5
+        SERIAL_NUMBER = 6
+        ACKNOWLEDGE = 7
+
+    def __init__(self, memcache, port='/dev/ttyUSB0', baudrate=9600, timeout=0.1, _version=1, **kwargs):
+        """Controller."""
+        self._memcache = memcache
+        self._serial = serial.Serial(port=port, baudrate=baudrate, timeout=timeout, **kwargs)
+        # Set default values, which should be overwritten quickly.
+        self.raw = b''
+        self.unit = Units.GRAINS
+        self.resolution = decimal.Decimal(0.02)
+        self.weight = decimal.Decimal('0.00')
+
+        self.StatusMap = self.ScaleStatusV1
+        if _version == 2:
+            self.StatusMap = self.ScaleStatusV2
+
+        self.status = self.StatusMap.STABLE
+        self.model_number = None
+        self.serial_number = None
+        atexit.register(self._graceful_exit)
+
+    def _graceful_exit(self):
+        """Graceful exit, closes serial port."""
+        logging.debug('Closing serial port...')
+        self._serial.close()
+
+    def change_unit(self):
+        """Changes the unit of weight on the scale."""
+        logging.debug('changing weight unit on scale from: %r', self.unit)
+        # Send Mode button command.
+        self._serial.write(b'\x1B\x73')
+        # Sleep 1s and wait for change to take effect.
+        time.sleep(1)
+        # Run update fn to set latest values.
+        self.update()
+
+    @property
+    def is_stable(self):
+        """Returns True if the scale is stable, otherwise False."""
+        return self.status == self.StatusMap.STABLE
+        # returns 1 when previous weights match with current
+        # return len(set(self._previous_reads)) == 1
+
+    def update(self):
+        """Read from the serial port and update an instance of this class with the most recent values."""
+        handlers = {
+            ' GN': self._stable,
+            '   ': self._unstable,
+            'F---.-H': self._overload,
+            None: noop,
+        }
+
+        # Note: The input buffer can fill up, causing latency. Clear it before reading.
+        self._serial.reset_input_buffer()
+        self._serial.write(b'\x1B\x70')
+        raw = self._serial.readline()
+        self.raw = raw
+        logging.debug(raw)
+        try:
+            line = raw.strip().decode('utf-8')
+        except UnicodeDecodeError:
+            logging.debug('Could not decode bytes to unicode.')
+        else:
+            status = line[9:12].strip()
+            fault = line[2:9].strip()
+            handler = handlers.get(status, noop)
+            handler2 = handlers.get(fault, noop)
+            handler(line)
+            handler2(line)
```

Comment:
There may be a better way to do this than calling both functions when only one is needed. I know that it's a little more complicated because of the data format. I don't have a suggestion off the top of my head.

At the very least, let's rename the functions to something more clear instead of `handler` and `handler2` and add some comments to document what is being done here and why.